lang/ruby/lib/avro/schema.rb

```diff
@@ -90,25 +90,43 @@ def self.real_parse(json_obj, names=nil, default_namespace=nil)
       end
     end
 
+    def self.is_integer?(datum)
+      (datum.is_a?(Fixnum) || datum.is_a?(Bignum)) && (INT_MIN_VALUE <= datum) && (datum <= INT_MAX_VALUE)
+    end
+
+    def self.is_long?(datum)
+      (datum.is_a?(Fixnum) || datum.is_a?(Bignum)) && (LONG_MIN_VALUE <= datum) && (datum <= LONG_MAX_VALUE)
+    end
+
+    def self.is_floating_point?(datum)
+      datum.is_a?(Float) || datum.is_a?(Fixnum) || datum.is_a?(Bignum)
+    end
+
+    def self.is_fixed?(datum, size)
+      datum.is_a?(String) && datum.size == size
+    end
+
+    def self.is_boolean?(datum)
+      datum == true || datum == false
+    end
+
```

Comment:
i'm somewhat unclear why you had to hoist these checks into individual functions -- couldn't you have simply reused `Schema.validate`?